client/src/pages/CreateAccount/CreateAccount.jsx

```diff
@@ -4,30 +4,65 @@ import LoggedOutNavBar from '../../components/LoggedOutNavBar/LoggedOutNavBar';
 import { useNavigate } from 'react-router-dom';
 import './CreateAccount.css';
 
-const CreateAccount = ({ onSubmit }) => {
+const CreateAccount = () => {
     const navigate = useNavigate();
 
-    const [email, setEmail] = useState('');
-    const [password, setPassword] = useState('');
-    const [confirmPassword, setConfirmPassword] = useState('');
-    const [name, setName] = useState('');
-    const [dobMonth, setDobMonth] = useState('');
-    const [dobDay, setDobDay] = useState('');
-    const [dobYear, setDobYear] = useState('');
-    const [gender, setGender] = useState('');
-    const [internOrNewGrad, setInternOrNewGrad] = useState('');
-    const [budgetMin, setBudgetMin] = useState('');
-    const [budgetMax, setBudgetMax] = useState('');
-    const [university, setUniversity] = useState('');
-    const [company, setCompany] = useState('');
-    const [groupId] = useState('');
+    const [formState, setFormState] = useState({
+        email: '',
+        password: '',
+        confirmPassword: '',
+        name: '',
+        dobMonth: '',
+        dobDay: '',
+        dobYear: '',
+        gender: '',
+        internOrNewGrad: '',
+        budgetMin: '',
+        budgetMax: '',
+        university: '',
+        company: '',
+        groupId: ''
+    });
+
     const [step, setStep] = useState(1);
     const [passwordError, setPasswordError] = useState('');
+    const [submitError, setSubmitError] = useState('');
+    const [isSubmitting, setIsSubmitting] = useState(false);
+
+    const createUser = async (userData) => {
+        try {
+            const response = await fetch('http://localhost:3000/api/users/create-account', {
```

Comment:
You should probably leave out the `localhost` so you can deploy this somewhere. Just make it `/api/users/create-account`, and the browser will work out the rest.